unit_tests/src/test_connection_pool.cpp

```diff
@@ -107,5 +113,129 @@ TEST_CASE("connection pool")
+            }
+        };
+
+        irods::connection_pool conn_pool{1,
+                                         env.rodsHost,
+                                         env.rodsPort,
+                                         proxy_admin.name,
+                                         env.rodsZone,
+                                         env.rodsUserName,
+                                         env.rodsZone,
+                                         600,
+                                         auth_func};
+
+        auto conn = conn_pool.get_connection();
+
+        // Show that the proxied connection works by accessing the home collection of
+        // the proxied user.
+        namespace fs = irods::experimental::filesystem;
+        REQUIRE(fs::client::exists(conn, fmt::format("/{}/home/{}", env.rodsZone, env.rodsUserName)));
+    }
+
+    SECTION("connection pool throws exception on incorrect alternative authentication method")
+    {
+        REQUIRE_THROWS([&env] {
+            const auto auth_func = [](auto& _comm)
+            {
+                // This will cause the connection pool to throw an exception on construction.
+                if (clientLoginWithPassword(&_comm, "nope") != 0) {
+                    THROW(SYS_SOCK_CONNECT_ERR, "Password authentication error");
+                }
+            };
+
+            irods::connection_pool conn_pool{1,
+                                             env.rodsHost,
+                                             env.rodsPort,
+                                             env.rodsUserName,
+                                             env.rodsZone,
+                                             600,
+                                             auth_func};
+        }());
+    }
+
+    SECTION("connection pool with proxy throws exception on incorrect alternative authentication method")
+    {
+        irods::experimental::client_connection admin_conn;
+
+        namespace ia = irods::experimental::administration;
+
+        const ia::user proxy_admin{"proxy_admin"};
+        REQUIRE_NOTHROW(ia::client::add_user(admin_conn, proxy_admin, ia::user_type::rodsadmin));
+
+        irods::at_scope_exit remove_proxy_admin{[&admin_conn, proxy_admin] {
+            ia::client::remove_user(admin_conn, proxy_admin);
+        }};
+
+        // The property is mutable so that we can use it for clientLoginWithPassword()
+        // later in the test.
+        auto password = std::to_array("ppass");
+        const ia::user_password_property pwd_property{password.data()};
+        REQUIRE_NOTHROW(ia::client::modify_user(admin_conn, proxy_admin, pwd_property));
+
+        const auto auth_func = [](auto& _comm)
+        {
+            // This will cause the connection pool to throw an exception on construction.
+            if (clientLoginWithPassword(&_comm, "nope") != 0) {
+                THROW(SYS_SOCK_CONNECT_ERR, "Password authentication error");
+            }
```

Comment:
For some reason, `SYS_SOCK_CONNECT_ERR` is not thrown. Instead, `CAT_INVALID_AUTHENTICATION` is thrown. I'm guessing this is due to the new authentication function related to the new authentication framework.

Reply:
It appears that `CAT_INVALID_AUTHENTICATION` is being printed to the terminal. It's a result of the `clientLogin*` functions using `fprintf`.

They should not be printing messages to the terminal directly.

I'll write up an issue for this.

Reply:
Adjusting the test proved that my assumptions about the `clientLogin*` functions and `fprintf`.

Reply:
Created #7048.

Done.